REVIEWING.md

```diff
@@ -0,0 +1,91 @@
+# Reviewing
+
+Pull request reviews are required before merging code into Vector. This document
+will outline Vector's pull request review requirements. The following checklist
+should be used for all pull requests:
+
+- [ ] Are you the code owner for the files that have changed? If not, please involve the appropriate code owner(s). (see [Code Owners](#code-owners))
+- [ ] Is the code addressing a single purpose? If not, the pull request should be broken up. (see [Single Purpose](#single-purpose))
+- [ ] Is the code readable and maintainable? If not, suggest ways to improve this. (see [Readability](#readability))
+- [ ] Is the code reasonably tested? If not, tests should be improved. (see [Testing](#testing))
+- [ ] Is code marked as unsafe? If so, verify that this is necessary. (see [Safe Code](#safe-code))
+- [ ] Is backward compatibility broken? If so, can it be avoided or deprecated? (see [Backwards compatibility](#backwards-compatibility))
+- [ ] Have dependencies changed? (see [Dependencies](#dependencies))
+- [ ] Has the code been explicitly reviewed for security issues? Dependencies included. (see [Security](#security))
+- [ ] Is there a risk of performance regressions? If so, have run the [Vector test harness](https://github.com/timberio/vector-test-harness)? (see [Performance Testing](#performance-testing))
+- [ ] Should documentation be adjusted to reflect any of these changes? (see [Documentation](#documentation))
+
+## Backwards Compatibility
+
+## Code Of Conduct
+
+If you have not, please review Vector's [Code of Conduct](CODE_OF_CONDUCT.md)
+to ensure reviews are welcoming, open, and respectful.
+
+## Code Owners
+
+Code owners should be involved in reviews that touch their code. This is
+defined in the [`.github/CODEOWNERS`](.github/CODE_OWNERS) file. Involving code
+owners ensures all context is applied to reviews since they have intimate
+knowledge of the code.
+
+## Dependencies
+
+Dependencies should be _carefully_ selected. Before adding a dependency we
+should ask the following questions:
+
+1. Is the dependency necessary? Can we avoid it?
```

Comment:
This seems a little strong. We could always avoid the dependency by rewriting it ourselves, so it's more a question of whether or not the benefit is worth the cost.